crates/client-api/src/lib.rs

```diff
@@ -20,13 +27,110 @@ pub mod util;
 ///
 /// Types returned here should be considered internal state and **never** be
 /// surfaced to the API.
+#[async_trait]
 pub trait NodeDelegate: Send + Sync {
     fn gather_metrics(&self) -> Vec<prometheus::proto::MetricFamily>;
-    fn host_controller(&self) -> &HostController;
     fn client_actor_index(&self) -> &ClientActorIndex;
 
     type JwtAuthProviderT: auth::JwtAuthProvider;
     fn jwt_auth_provider(&self) -> &Self::JwtAuthProviderT;
+    /// Return the leader [`Host`] of `database_id`.
+    ///
+    /// Returns `None` if the current leader is not hosted by this node.
+    /// The [`Host`] is spawned implicitly if not already running.
+    async fn leader(&self, database_id: u64) -> anyhow::Result<Option<Host>>;
+    fn module_logs_dir(&self, replica_id: u64) -> ModuleLogsDir;
```

Comment:
How do you feel about adding the following to `Host` instead:

```rust
pub async fn module_logs(&self, num_lines: Option<u32>) -> String {
    let logs_dir = self.host_controller.data_dir.replica(self.replica_id).module_logs();
    DatabaseLogger::read_latest(logs_dir, num_lines).await
}
```

Reply:
(Not sure why we're reading all of those into a single big string instead of a stream of lines, but that'd be for another time)

Reply:
> How do you feel about adding the following to Host instead:

I think we want to fetch these module_logs without actually launching host if it is not already up, actually depends on `follow` - https://github.com/clockworklabs/SpacetimeDB/blob/master/crates/client-api/src/routes/database.rs#L457